packages/password-input/src/PasswordInput/PasswordInput.tsx

```diff
@@ -0,0 +1,208 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+
+import { cx } from '@leafygreen-ui/emotion';
+import { useControlledValue, useIdAllocator } from '@leafygreen-ui/hooks';
+import LeafyGreenProvider, {
+  useDarkMode,
+} from '@leafygreen-ui/leafygreen-provider';
+import { Label } from '@leafygreen-ui/typography';
+
+import { InputIcon } from '../InputIcon';
+import { StateNotifications } from '../StateNotifications';
+import { TogglePassword } from '../TogglePassword';
+
+import {
+  inputBaseStyles,
+  inputBaseThemeStyles,
+  inputDisabledBaseStyles,
+  inputDisabledThemeStyles,
+  inputIconSizeStyles,
+  inputSizeStyles,
+  inputThemeStyles,
+  inputWrapperStyles,
+  labelBaseStyles,
+  labelLargeOverrideStyles,
+} from './PasswordInput.styles';
+import {
+  type NotificationProps,
+  PasswordInputProps,
+  SizeVariant,
+  States,
+} from './PasswordInput.types';
+import { getStateFromArray } from './utils';
+
+export const PasswordInput = React.forwardRef<
+  HTMLInputElement,
+  PasswordInputProps
+>(
+  (
+    {
+      value: valueProp,
+      onChange: onChangeProp,
+      className,
+      darkMode: darkModeProp,
+      label,
+      sizeVariant = SizeVariant.Default,
+      stateNotifications = [],
+      disabled = false,
+      autoComplete = 'new-password',
+      id: idProp,
+      'aria-describedby': ariaDescribedbyProp,
+      'aria-labelledby': ariaLabelledbyProp,
+      ...rest
+    }: PasswordInputProps,
+    forwardedRef,
+  ) => {
+    const [showPassword, setShowPassword] = useState<boolean>(false);
+    const prefix = 'passwordinput';
+    const inputId = useIdAllocator({ prefix, id: idProp });
+    const ariaDescribedby = useIdAllocator({
+      prefix,
+      id: ariaDescribedbyProp,
+    });
+    const ariaLabelledby = useIdAllocator({
```

Comment:
Does this work here? I think if they pass an aria-labelledby we don't render a label element because their id is pointing to something already in the DOM

Reply:
if they pass in `aria-labelledby` that will be used instead of generating an ID, the label will not render, and the id will be passed to the `input` element with `aria-describedby`.